src/main/java/net/querz/nbt/CompoundTag.java

```diff
@@ -458,16 +450,14 @@ public CompoundTag getCompoundTag(String key) {
 		return null;
 	}
 
-	public boolean contains(String key, int type) {
+	public boolean contains(String key, Type type) {
 		Tag tag = value.get(key);
-		byte t = tag == null ? END : tag.getID();
-		if (t == type) {
-			return true;
-		} else if (type != NUMBER) {
-			return false;
-		} else {
-			return t >= BYTE && t <= DOUBLE;
-		}
+		return tag.getType() == type;
```

Comment:
this will throw an NPE if `value` does not contain this key

Reply:
Oh wow, not sure how I missed that here but did `containsNumber` correctly...